File: C Modules/Assignments/Assignment no 9/Assignment no 7/Question-2/Assignment no 3/Type-3/Question_8.c
#include<stdio.h>
int strong(int num[10]);

int main()
{
    int num[10];
    int result=strong(num);
    int *ptr2=&result;
    if (*ptr2==1)
    {
        printf("strong number");
    }
    else{
        printf("not a strong number");
    }
    return 0;
}

int strong(int num[10])
{
    int temp,digit,sum=0,fact,i;
    int *ptr1=&num[0];
    printf("Enter the number:");
    scanf(" %d", ptr1);
    temp=*ptr1;
    while (temp>0)
    {
        digit=temp%10;
        fact=1;
        i=1;
        while (i<=digit)
        {
            fact=fact*i;
            i++;
        }
        sum=sum+fact;
        temp=temp/10;

    }
    if (sum==*ptr1)
    {
        return 1;
    }
    else{
        return 0;
    }
}

File: Java Module/Assignments in c to java/Assignment-03/Question_8.java
public class Question_8 {

public static void main(String[] args) {
        int num=145,temp,digit,sum=0,fact,i;
    // printf("Enter the number\n");
    // scanf(" %d", &num);
    temp=num;
    while (temp>0)
    {
        digit=temp%10;
        fact=1;
        i=1;
        while (i<=digit)
        {
            fact=fact*i;
            i++;
            /* code */
        }
        sum=sum+fact;
        temp=temp/10;
        /* code */
    }if (sum==num)
    {
        System.out.printf("strong number: %d",num);
        /* code */
    }
    else{
        System.out.printf("not a strong number: %d",num);
    }
}
}